q2cli/core/usage.py

```diff
@@ -0,0 +1,156 @@
+        params_t = self._template_parameters(params, input_opts)
+        mds_t = self._template_metadata(mds, input_opts)
+        outputs_t = self._template_outputs(action_sig, outputs)
+        t = " \\\n".join([cmd] + inputs_t + params_t + mds_t + outputs_t)
+        return t
+
+    def _template_inputs(self, action_sig, input_opts):
+        inputs = []
+        for i in action_sig.inputs:
+            if i in input_opts:
+                p = f"--i-{to_cli_name(i)}"
+                val = f"{input_opts[i]}.qza"
+                inputs.append(f"{' ':>4}{p} {val}")
+        return inputs
+
+    def _template_parameters(self, params, input_opts):
+        params_t = []
+        for i, spec in params:
+            val = str(input_opts[i]) if i in input_opts else ""
+            if spec.qiime_type is Bool:
+                pfx = f"--p-" if val == "True" else f"--p-no-"
+                p = f"{pfx}{to_cli_name(i)}"
+                params_t.append(f"{' ':>4}{p}")
+            elif val:
+                p = f"--p-{to_cli_name(i)}"
+                _val = f" {val}"
+                params_t.append(f"{' ':>4}{p + _val}")
+        return params_t
+
+    def _template_outputs(self, action_sig, outputs):
+        outputs_t = []
+        for i, spec in action_sig.outputs.items():
+            qtype = spec.qiime_type
+            ext = ".qzv" if is_visualization_type(qtype) else ".qza"
+            p = f"--o-{to_cli_name(i)}"
+            val = f"{to_snake_case(outputs[i])}{ext}"
+            outputs_t.append(f"{' ':>4}{p} {val}")
+        return outputs_t
+
+    def _template_metadata(self, mds, input_opts):
+        mds_t = []
+        data = self.get_example_data()
+        data = {k: v for k, v in data.items() if isinstance(v, Metadata)}
+        file_param = f"--m-metadata-file"
+        col_param = f"--m-metadata-column"
+        for i, spec in mds:
+            qtype = spec.qiime_type
+            if not is_metadata_column_type(qtype):
+                name = str(input_opts[i]) if i in input_opts else ""
+                if name in data:
+                    # Metadata item `i` was registered by Usage.init_data and
+                    # can therefore be assumed *not* to be a merge target.
+                    mds_t.append(f"{' ':>4}{file_param} {name}.tsv")
+                else:
+                    # Metadata item `i` was not registered by Usage.init_data
+                    # and can therefore be assumed to be a merge target.
+                    #
+                    # Extract metadata that was merged into merge target `i`
+                    for k, v in data.items():
+                        mds_t.append(f"{' ':>4}{file_param} {k}.tsv")
+            elif is_metadata_column_type(qtype):
+                col, md = input_opts[i]
+                mds_t.append(f"{' ':>4}{file_param} {md}.tsv")
+                mds_t.append(f"{' ':>4}{col_param} '{col}'")
+        return mds_t
+
+
+def examples(action):
```

Comment:
This should be refactored to use the cached values, now that you have that available to you.

Reply:
If I'm not mistaken, I think the the click option is the only place that needs to know about the cache.  Usage examples will only be rendered via `q2cli.core.usage.examples` when the cache is refreshed as per the pre-existing logic.  See my inline comment in `q2cli/util.py`